code/modules/antagonists/ert/idpd.dm

```diff
@@ -0,0 +1,175 @@
+	. = ..()
+	if(!visualsOnly)
+		H.dna.add_mutation(TK)
+/datum/outfit/ert/idpd/titan
+	name = "I.D.P.D Titan"
+
+	suit = /obj/item/clothing/suit/space/hardsuit/ert/idpd/titan
+	l_pocket = null
+	l_hand = /obj/item/pickaxe/drill/jackhammer //BREACHING
+	r_hand = null
+
+/datum/outfit/ert/idpd/titan/post_equip(mob/living/carbon/human/H, visualsOnly)
+	. = ..()
+	if(visualsOnly)
+		return
+	var/obj/effect/proc_holder/spell/targeted/forcewall/idpd/wall = new
+	H.AddSpell(wall)
+
+
+////Antag Datums////
+/datum/antagonist/ert/idpd
+	name = "I.D.P.D Grunt"
+	role = "I.D.P.D Grunt"
+	outfit = /datum/outfit/ert/idpd
+
+/datum/antagonist/ert/idpd/gazer
+	name = "I.D.P.D Gazer"
+	role = "I.D.P.D Gazer"
+	outfit = /datum/outfit/ert/idpd/gazer
+
+/datum/antagonist/ert/idpd/titan
+	name = "I.D.P.D Titan"
+	role = "I.D.P.D Titan"
+	outfit = /datum/outfit/ert/idpd/titan
+
+/datum/antagonist/ert/idpd/chief
+	name = "I.D.P.D Chief"
+	role = "I.D.P.D Chief"
+	outfit = /datum/outfit/ert/idpd/chief
+
+////Titan Spell
+/obj/effect/proc_holder/spell/targeted/forcewall/idpd
+	name = "Barrier Projector"
+	desc = "Create an energy barrier to protect your team."
+	charge_max = 400
+	invocation = ""
+	invocation_type = "none"
+	wall_type = /obj/effect/forcefield/idpd
+
+/obj/effect/forcefield/idpd
+
+/obj/effect/forcefield/idpd/CanPass(atom/movable/mover, turf/target)
+	return FALSE //None shall pass
+
+////IDPD Creation, handled by one_click_antag.dm
+/datum/admins/proc/attemptMakeIDPD(var/list/mob/dead/observer/candidates, var/num_agents, var/datum/ert/ertemplate, var/datum/team/ert/ert_team, var/datum/objective/missionobj) //They don't start at centcom (fuck you linter)
+	if(num_agents > candidates.len || candidates.len == 0)
+		return FALSE
+
+	var/list/mob/dead/observer/chosen = list()
+	for(var/X = 1, X <= num_agents, ++X)
+		var/mob/dead/observer/chosen_one = pick(candidates)
+		candidates -= chosen_one
+		chosen += chosen_one
+
+	var/turf/spawnpoint = get_turf(pick(GLOB.blobstart)) //Ensures the portal spawns on station... for the most part.
+	var/obj/effect/portal/idpd_portal = new(spawnpoint.loc) //"Fake" portal spawns 20 seconds before the IDPD spawns in
+
+	message_admins("I.D.P.D portal spawned at [ADMIN_VERBOSEJMP(idpd_portal)]")
+
+	for(var/mob/dead/observer/chosen_one in chosen)
+		chosen_one.orbit(idpd_portal)
+
+	priority_announce("FLÄSHYN, ÖHU!", "Higher Dimensional Affairs", ANNOUNCER_SPANOMALIES)
+	playsound(spawnpoint, 'sound/misc/idpd_portal.ogg', 100, 1)
+	addtimer(CALLBACK(src, .proc/spawnIDPD, chosen, ertemplate, idpd_portal, ert_team, missionobj), 20 SECONDS)
+	return TRUE
+
+/datum/admins/proc/spawnIDPD(var/list/mob/dead/observer/chosen, var/datum/ert/ertemplate, var/obj/effect/portal/idpd_portal, var/datum/team/ert/ert_team, var/datum/objective/missionobj)
+	var/spawnloc = idpd_portal.loc
+	while(chosen.len)
+		var/mob/dead/observer/newguy = pick(chosen)
+		chosen -= newguy
+		if(!newguy.key)
+			continue
```

Comment:
Why would this ever happen?

Reply:
Don't ask me. It was taken from E.R.T code.